core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/io/FastDoubleParser.kt

```diff
@@ -0,0 +1,233 @@
+         * [additionals] will be added to the set too, when they're safe to add.
+         */
+        fun ((DecimalFormatSymbols) -> String).fromLocalWithFallBack(vararg additionals: String): Set<String> =
+            buildSet {
+                val getString = this@fromLocalWithFallBack
+                val string = getString(localDecimalFormatSymbols).lowercase()
+                add(string)
+
+                // add fallback string if it's safe to do so
+                val fallbackString = getString(fallbackDecimalFormatSymbols).lowercase()
+                if (!fallbackString.any { it in localChars } && fallbackString !in localStrings) {
+                    add(fallbackString)
+                }
+
+                // Fixes NBSP and other whitespace characters not being recognized if the user writes space instead.
+                if (string.isBlank()) add(" ")
+
+                // add additional strings if needed
+                for (additional in additionals) {
+                    val lowercase = additional.lowercase()
+                    if (!lowercase.any { it in localChars } && lowercase !in localStrings) {
+                        add(lowercase)
+                    }
+                }
+            }
+
+        return NumberFormatSymbols.fromDecimalFormatSymbols(localDecimalFormatSymbols)
+            .withPlusSign(setOf('+'))
+            .withDecimalSeparator(DecimalFormatSymbols::getDecimalSeparator.fromLocalWithFallBack())
+            .withGroupingSeparator(DecimalFormatSymbols::getGroupingSeparator.fromLocalWithFallBack())
+            .withExponentSeparator(DecimalFormatSymbols::getExponentSeparator.fromLocalWithFallBack())
+            .withMinusSign(DecimalFormatSymbols::getMinusSign.fromLocalWithFallBack())
+            .withInfinity(DecimalFormatSymbols::getInfinity.fromLocalWithFallBack(*INFINITIES))
+            .withNaN(DecimalFormatSymbols::getNaN.fromLocalWithFallBack(*NANS))
+    }
+
+    /** Fallback method for parsing doubles. */
+    private fun String.parseToDoubleOrNullFallback(): Double? =
+        when (lowercase()) {
+            in INFINITIES, in PLUS_INFINITIES -> Double.POSITIVE_INFINITY
+
+            in MINUS_INFINITIES -> Double.NEGATIVE_INFINITY
+
+            in NANS -> Double.NaN
+
+            else -> {
+                // not thread safe; must be created here
+                val numberFormat = NumberFormat.getInstance(locale)
+                val parsePosition = ParsePosition(0)
+                val result = numberFormat.parse(this, parsePosition)?.toDouble()
+                if (parsePosition.index != this.length || parsePosition.errorIndex != -1) {
+                    null
+                } else {
+                    result
+                }
+            }
+        }.also {
+            if (it == null) {
+                logger.debug { "Could not parse '$this' as Double with NumberFormat with locale '$locale'." }
```

Comment:
Probably we could also add some logs about **experimental fast double parsing** like a marker for our logs

Reply:
KLogger already provides a marker. It recognizes the file/class it's called in. So this will look something like: `DEBUG org.jetbrains.kotlinx.dataframe.impl.io.FastDoubleParser - Could not parse 'hello' as Double with NumberFormat with locale 'en_US'.`

Reply:
we can just search for "FastDoubleParser" :)